src/Toolkit/Toolkit/UI/Controls/BasemapGallery/BasemapGalleryController.cs

```diff
@@ -283,55 +283,58 @@ private static bool BasemapIsActuallyNotABasemap(Basemap input)
             return false;
         }
 
-        private static async Task<IList<BasemapGalleryItem>?> PopulateBasemapsForPortal(ArcGISPortal? portal)
+        private async Task<IList<BasemapGalleryItem>?> PopulateBasemapsForPortal(ArcGISPortal? portal)
         {
-            if (portal == null)
-            {
-                return null;
-            }
+            if (portal == null) return null;
 
-            Task<IEnumerable<Basemap>> getBasemapsTask;
-            if (portal.PortalInfo?.UseVectorBasemaps ?? false)
-            {
-                getBasemapsTask = portal.GetVectorBasemapsAsync();
-            }
-            else
-            {
-                getBasemapsTask = portal.GetBasemapsAsync();
-            }
+            await LoadBasemapGalleryItems(portal);
+            return GetBasemapGalleryItems();
+        }
 
-            var basemaps = await getBasemapsTask;
+        private async Task<IList<BasemapGalleryItem>> PopulateFromDefaultList(CancellationToken cancellationToken = default)
+        {
+            ArcGISPortal defaultPortal = await ArcGISPortal.CreateAsync(cancellationToken);
 
-            IList<BasemapGalleryItem> listOfBasemaps = new List<BasemapGalleryItem>();
+            await LoadBasemapGalleryItems(defaultPortal, cancellationToken);
+            return GetBasemapGalleryItems();
+        }
 
-            foreach (var item in basemaps)
+        private ObservableCollection<BasemapGalleryItem> GetBasemapGalleryItems()
```

Comment:
I would merge this with LoadBasemapGalleryItems, so there's no mistake in order of operations. You can return the basemaps if they are already there, or load them first to cache then return them. ie something like
```
if(_cached3DBasemaps is null)
  _cached3DBasemaps = await Getbasemaps...()
basemapGalleryItems.AddRange(_cached3DBasemaps);
```

Also make sure to clear these caches if the portal ever changes.